policy/common/methods.go

```diff
@@ -120,9 +120,11 @@ func deduplicateCandidates(all []*Candidate) []*Candidate {
 }
 
 func (m *Methods) CommentMatches(commentBody string) bool {
-	for _, comment := range m.Comments {
-		if strings.Contains(commentBody, comment) {
-			return true
+	if m.Comments != nil {
```

Comment:
Same idea here with the nil check and dereferencing: you can `range` over a `nil` slice without checking first.